File: data-structures/one-dimensional-arrays/python/array_loop.py
# Isaac Computer Science
# Usage licensed under the Open Government Licence v3.0


def main():
    """Create a list and output a value using an index value"""
    emotions = ["amazed", "delighted", "ecstatic", "enthusiastic", "lively"]

    print(emotions[2])  # Displays the third item in the list
    
    emotions.append("happy")  # Appends an item to the end of the list
    
    emotions.pop(1)  # Removes the second item from the list

    list_length = len(emotions)  # Get length of list
    for i in range(list_length):
        print(emotions[i])


# This code will run if this file is executed directly
# (i.e. not called by another program)
if __name__ == '__main__':
    main()

File: data-structures/one-dimensional-arrays/c-sharp/array_loop.cs
/*
Raspberry Pi Foundation
Developed to be used alongside Isaac Computer Science, part of the National Centre for Computing Education
Usage licensed under CC BY-SA 4
Note: This file is designed to be copied out and compiled on your machine.
In order for it to compile properly you need to ensure that the project name is the same as the "namespace" in this file.
To run this file you need to:
1. Copy the contents
2. Paste them into the C# IDE of your choice (Visual Studio, for example)
3. Change the namespace to match your project (if necessary)
4. Compile the program
5. Run the program
*/

using System;

namespace IsaacCodeSamples
{

    class OneDimensionalArrays
    {
        // The Main method is the entry point for all C# programs
        public static void Main()
        {
            string[] emotions = new string[6]; // Declare an array of 6 elements

            emotions[0] = "amazed";  // Assign a value into a position of the array
            emotions[1] = "delighted";
            emotions[2] = "ecstatic";
            emotions[3] = "enthusiastic";
            emotions[4] = "lively";

            Console.WriteLine(emotions[2]);  // Displays the third element of the array

            emotions[5] = "happy";  // Adds a new item to the array

            emotions[1] = "";  // Deletes the item in the second element of the array

            for (int i = 0; i < 6; i++)
            {
                Console.WriteLine(emotions[i]);
            }

        }
    }

}
